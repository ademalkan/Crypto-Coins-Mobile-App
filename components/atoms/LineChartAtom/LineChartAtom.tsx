import React from 'react';
import { LineChart } from 'react-native-chart-kit';

interface LineChartProps {
  data: {
    labels: string[];
    datasets: {
      data: number[];
    }[];
  };
  width: number;
  height: number;
}

const LineChartAtom: React.FC<LineChartProps> = ({ data, width, height }) => {
  return (
    <LineChart
      data={data}
      width={width}
      height={height}
      chartConfig={{
        backgroundColor: '#ffffff',
        backgroundGradientFrom: '#ffffff',
        backgroundGradientTo: '#ffffff',
        decimalPlaces: 2,
        color: (opacity = 1) => `rgba(0, 0, 0, ${opacity})`,
        style: {
          borderRadius: 8,
        },
      }}
      bezier
    />
  );
};

export default LineChartAtom;
