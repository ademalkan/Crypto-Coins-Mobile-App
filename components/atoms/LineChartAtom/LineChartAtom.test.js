import React from "react";
import renderer from "react-test-renderer";
import LineChartAtom from "./LineChartAtom";

describe("LineChartAtom", () => {
  it("should render the LineChart component correctly", () => {
    const data = {
      labels: ["January", "February", "March", "April", "May", "June"],
      datasets: [
        {
          data: [20, 45, 28, 80, 99, 43],
        },
      ],
    };
    const width = 300;
    const height = 200;

    const component = renderer.create(
      <LineChartAtom data={data} width={width} height={height} />
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
