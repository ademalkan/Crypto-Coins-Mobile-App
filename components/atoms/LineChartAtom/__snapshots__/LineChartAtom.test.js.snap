// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChartAtom should render the LineChart component correctly 1`] = `
<View
  style={{}}
>
  <RNSVGSvgView
    bbHeight="200"
    bbWidth="300"
    focusable={false}
    height={200}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 200,
          "width": 300,
        },
      ]
    }
    width={300}
  >
    <RNSVGGroup>
      <RNSVGRect
        fill={
          {
            "brushRef": "backgroundGradient",
            "type": 1,
          }
        }
        fillOpacity={1}
        height="200"
        propList={
          [
            "fill",
            "fillOpacity",
          ]
        }
        rx="0"
        ry="0"
        width="100%"
        x="0"
        y="0"
      />
      <RNSVGGroup
        matrix={
          [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              [
                0,
                -1,
                1,
                -1,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="backgroundGradient"
            x1="0"
            x2="300"
            y1="200"
            y2="0"
          />
          <RNSVGLinearGradient
            gradient={
              [
                0,
                436207616,
                1,
                436207616,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="fillShadowGradientFrom"
            x1="0"
            x2="0"
            y1="0"
            y2="200"
          />
        </RNSVGDefs>
        <RNSVGGroup>
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="300"
            y1="16"
            y2="16"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="300"
            y1="53.5"
            y2="53.5"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="300"
            y1="91"
            y2="91"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="300"
            y1="128.5"
            y2="128.5"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="300"
            y1="166"
            y2="166"
          />
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "end",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "52",
              ]
            }
            y={
              [
                "166",
              ]
            }
          >
            <RNSVGTSpan
              content="20.00"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "end",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "52",
              ]
            }
            y={
              [
                "128.5",
              ]
            }
          >
            <RNSVGTSpan
              content="39.75"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "end",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "52",
              ]
            }
            y={
              [
                "91",
              ]
            }
          >
            <RNSVGTSpan
              content="59.50"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "end",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "52",
              ]
            }
            y={
              [
                "53.5",
              ]
            }
          >
            <RNSVGTSpan
              content="79.25"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "end",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "52",
              ]
            }
            y={
              [
                "16",
              ]
            }
          >
            <RNSVGTSpan
              content="99.00"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="64"
            x2="64"
            y1="0"
            y2="166"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="103"
            x2="103"
            y1="0"
            y2="166"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="142"
            x2="142"
            y1="0"
            y2="166"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="182"
            x2="182"
            y1="0"
            y2="166"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="221"
            x2="221"
            y1="0"
            y2="166"
          />
          <RNSVGLine
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeDasharray",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeDasharray={
              [
                "5",
                "10",
              ]
            }
            strokeWidth="1"
            x1="260"
            x2="260"
            y1="0"
            y2="166"
          />
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "64",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="January"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "103.33333333333334",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="February"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "142.66666666666669",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="March"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "182",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="April"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "221.33333333333334",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="May"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={[]}
            dy={[]}
            fill={
              {
                "payload": 3422552064,
                "type": 0,
              }
            }
            font={
              {
                "fontSize": "12",
                "textAnchor": "middle",
              }
            }
            matrix={
              [
                1,
                0,
                -0,
                1,
                0,
                0,
              ]
            }
            propList={
              [
                "fill",
              ]
            }
            rotate={[]}
            x={
              [
                "260.6666666666667",
              ]
            }
            y={
              [
                "190",
              ]
            }
          >
            <RNSVGTSpan
              content="June"
              dx={[]}
              dy={[]}
              font={{}}
              rotate={[]}
              x={[]}
              y={[]}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGPath
            d="M64,166 Q 73.75, 166, 83.5, 142 Q 93.25, 118, 103, 118 Q 112.75, 118, 122.5, 134 Q 132.25, 150, 142, 150 Q 152, 150, 162, 101 Q 172, 52, 182, 52 Q 191.75, 52, 201.5, 34 Q 211.25, 16, 221, 16 Q 230.75, 16, 240.5, 69 Q 250.25, 122, 260, 122"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              {
                "payload": 855638016,
                "type": 0,
              }
            }
            strokeWidth="3"
          />
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGPath
            d="M64,166 Q 73.75, 166, 83.5, 142 Q 93.25, 118, 103, 118 Q 112.75, 118, 122.5, 134 Q 132.25, 150, 142, 150 Q 152, 150, 162, 101 Q 172, 52, 182, 52 Q 191.75, 52, 201.5, 34 Q 211.25, 16, 221, 16 Q 230.75, 16, 240.5, 69 Q 250.25, 122, 260, 122 L260.6666666666667,166 L64,166 Z"
            fill={
              {
                "brushRef": "fillShadowGradientFrom",
                "type": 1,
              }
            }
            propList={
              [
                "fill",
                "strokeWidth",
              ]
            }
            strokeWidth="0"
          />
        </RNSVGGroup>
        <RNSVGGroup>
          <RNSVGCircle
            cx="64"
            cy="166"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="64"
            cy="166"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
          <RNSVGCircle
            cx="103.33333333333334"
            cy="118.53164556962025"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="103.33333333333334"
            cy="118.53164556962025"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
          <RNSVGCircle
            cx="142.66666666666669"
            cy="150.81012658227849"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="142.66666666666669"
            cy="150.81012658227849"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
          <RNSVGCircle
            cx="182"
            cy="52.0759493670886"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="182"
            cy="52.0759493670886"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
          <RNSVGCircle
            cx="221.33333333333334"
            cy="16"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="221.33333333333334"
            cy="16"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
          <RNSVGCircle
            cx="260.66666666666663"
            cy="122.32911392405065"
            fill={
              {
                "payload": 3858759680,
                "type": 0,
              }
            }
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
              ]
            }
            r="4"
            responsible={true}
          />
          <RNSVGCircle
            cx="260.66666666666663"
            cy="122.32911392405065"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            fillOpacity={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              [
                "fill",
                "fillOpacity",
              ]
            }
            r="14"
            responsible={true}
          />
        </RNSVGGroup>
        <RNSVGGroup />
        <RNSVGGroup />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
